src/nvim/api/buffer.c

```diff
@@ -757,6 +757,85 @@ void nvim_buf_set_text(uint64_t channel_id, Buffer buffer, Integer start_row, In
   try_end(err);
 }
 
+/// Gets a range from the buffer.
+///
+/// This differs from |nvim_buf_get_lines()| in that it allows retrieving only
+/// portions of a line.
+///
+/// Indexing is zero-based and end-exclusive.
+///
+/// Prefer |nvim_buf_get_lines()| when retrieving entire lines.
+///
+/// @param channel_id
+/// @param buffer           Buffer handle, or 0 for current buffer
+/// @param start_row        First line index
+/// @param start_column     First column
+/// @param end_row          Last line index
+/// @param end_column       Last column
+/// @param[out] err         Error details, if any
+ArrayOf(String) nvim_buf_get_text(uint64_t channel_id, Buffer buffer,
+                                  Integer start_row, Integer start_col,
+                                  Integer end_row, Integer end_col, Error *err)
+  FUNC_API_SINCE(8)
+{
+  Array rv = ARRAY_DICT_INIT;
+
+  // nvim_buf_get_lines doesn't care if the start row is greater than the end
+  // row (it will just return an empty array), but nvim_buf_get_text does in
+  // order to maintain symmetry with nvim_buf_set_text.
+  if (start_row > end_row || (end_row == start_row && start_col > end_col)) {
+    api_set_error(err, kErrorTypeValidation, "start is higher than end");
+    return rv;
+  }
+
+  Array lines = nvim_buf_get_lines(channel_id, buffer, start_row, end_row + 1,
```

Comment:
this still makes a memcpy of the full extent of the first and last line. Probably ok for now but could add a TODO?

Reply:
Removed in 60a7ad4